content/latest/api/ysql/commands/dml_insert.md

```diff
@@ -49,17 +69,32 @@ Where
 - `column_names` is a comma-separated list of columns names (identifiers).
 
 ## Semantics
+
+The bare INSERT variant is straightforward.
+
  - An error is raised if the specified table does not exist. 
- - Each of primary key columns must have a non-null value.
+ - Each of the primary key columns must have a non-null value.
 
 ### `VALUES` Clause
  - Each of the values list must have the same length as the columns list.
  - Each value must be convertible to its corresponding (by position) column type.
  - Each value literal can be an expression.
 
+The INSERT ON CONFLICT variant needs some explanation. The target table must have
+at least one column (list) with either a unique index or a unique constraint. We
+shall refer to this as a unique key. The argument of VALUES is a relation that must
+include at least one of the target table's unique keys. Some of the values of this
+unique key might be new, and others might already exist in the target table. The
+basic aim of INSERT ON CONFLICT is simply to insert the rows with new values of
+the unique key and to update the rows with existing values of the unique key to
+set the values of the remaining specified columns to those in the VALUES relation.
+In this way, the net effect is either to insert or to update; and for this reason
+the INSERT ON CONFLICT variant is often colloquially referred to as "upsert".
+
```

Comment:
Break this into two different paragraph.  The first one speaks of ON CONFLICT in general, and the second speaks of UPSERT.
```
- The target table must have ...
- The basic aim of INSERT ON CONFLICT ...
```